src/test/java/me/karakelley/http/RequestTest.java

```diff
@@ -2,78 +2,113 @@
+import java.util.Map;
+
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class RequestTest {
   Request request;
 
   @Test
   void testGetMethod() {
-    request = new Request(HttpMethod.GET, null, null, null, "".getBytes(), 0);
+    request = basicGetRequest();
+
     assertEquals(HttpMethod.GET, request.getMethod());
   }
 
   @Test
   void testGetPath() {
-    request = new Request(null, "/", null, null, "".getBytes(), 0);
+    request = basicGetRequest();
+
     assertEquals("/", request.getPath());
   }
 
   @Test
   void testGetProtocol() {
-    request = new Request(null, null, "HTTP/1.1", null, "".getBytes(), 0);
+    request = basicGetRequest();
+
     assertEquals("HTTP/1.1", request.getProtocol());
   }
 
   @Test
   void testGetHeaders() {
-    HashMap headers = new HashMap();
-    headers.put("Host", "localhost:5000");
-    headers.put("Connection", "keep-alive");
+    Map<String, String> headers = setHeaders();
+    request = basicGetRequest();
 
-    request = new Request(HttpMethod.GET, "/", "HTTP/1.1", headers, "".getBytes(), 0);
     assertEquals(headers, request.getHeaders());
   }
 
   @Test
   void testOneHeader() {
-    HashMap headers = new HashMap();
-    headers.put("Host", "localhost:5000");
-    headers.put("Connection", "keep-alive");
-    request = new Request(null, null, null, headers, "".getBytes(), 0);
+    request = basicGetRequest();
+
     assertEquals("localhost:5000", request.getHeader("Host"));
   }
 
   @Test
   void testBadRequest() {
     try {
-      request = new Request(HttpMethod.GET, "/", null, null, "".getBytes(), 0);
-    } catch (Exception e) {
+      request = new Request.Builder().setMethod(HttpMethod.GET).setPath("/").setPort(0).build();
+      } catch (Exception e) {
       assertEquals("java.lang.ArrayIndexOutOfBoundsException: 2", e.getMessage());
     }
   }
 
   @Test
   void testGetHostAndPort() {
-    request = new Request(HttpMethod.GET, "/",  "HTTP/1.1", new HashMap<>(), "".getBytes(),  4000);
-    assertEquals(request.getHostAndPort(), "localhost:4000" );
+    request = basicGetRequest();
+    assertEquals(request.getHostAndPort(), "localhost:5000" );
   }
 
   @Test
   void testGetHostAndPortWithHostDefined() {
-    HashMap headers = new HashMap();
+    HashMap<String, String> headers = new HashMap<>();
     headers.put("Host", "test.com");
-    request = new Request(HttpMethod.GET, "/",  "HTTP/1.1", headers, "".getBytes(), 4000);
+    request = basicGetRequest(headers, 4000);
     assertEquals(request.getHostAndPort(), "test.com:4000" );
   }
 
   @Test
-  void testgetBody() throws IOException {
-    HashMap headers = new HashMap();
-    headers.put("Content-Length", 9);
-    Request request = new Request(HttpMethod.POST, "/test1.txt", "HTTP/1.1", headers, "txt=99999".getBytes(), 0);
+  void testGetBody() {
+    HashMap<String, String> headers = new HashMap<>();
+    headers.put("Content-Length", "9");
+    Request request = new Request.Builder()
```

Comment:
I like this "method-chaining" pattern. Nice job. I will say that sometimes you can get really unhelpful stack traces in your logs from long method chains like this. If `setPort` throws a `NullPointerException` (for some wat Java `Integer` -> `int` implicit conversion reason) it will likely tell you that line 76 threw the exception. I still do it, though 🥇